packages/components/src/date-time/time.js

```diff
@@ -120,46 +180,127 @@ class TimePicker extends Component {
-						isToggled={ am === 'PM' }
-						onClick={ this.updateAmPm( 'PM' ) }
-					>
-						{ __( 'PM' ) }
-					</Button>
-				</div> }
-			</div>
+			<fieldset
+				className={
+					is12Hour ? 'components-datetime__time components-datetime__time--12-hour' : 'components-datetime__time components-datetime__time--24-hour'
+				}
+			>
+				<legend>{ __( 'Publish at' ) }</legend>
+				<fieldset className="invisible">
+					<legend className="invisible">{ __( 'Date' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--month">
+							<select
+								aria-label={ __( 'Month' ) }
+								name="month"
+								value={ month }
+								onChange={ this.onChangeMonth }
+								onBlur={ this.updateMonth }
+							>
+								<option value="01">{ __( 'January' ) }</option>
+								<option value="02">{ __( 'February' ) }</option>
+								<option value="03">{ __( 'March' ) }</option>
+								<option value="04">{ __( 'April' ) }</option>
+								<option value="05">{ __( 'May' ) }</option>
+								<option value="06">{ __( 'June' ) }</option>
+								<option value="07">{ __( 'July' ) }</option>
+								<option value="08">{ __( 'August' ) }</option>
+								<option value="09">{ __( 'September' ) }</option>
+								<option value="10">{ __( 'October' ) }</option>
+								<option value="11">{ __( 'November' ) }</option>
+								<option value="12">{ __( 'December' ) }</option>
+							</select>
+						</div>
+						<div key="day-field" className="components-datetime__time__form__container components-datetime__time__form__container--day">
+							<input
+								aria-label={ __( 'Day' ) }
+								type="number"
+								name="day"
+								value={ day }
+								step={ 1 }
+								min={ 1 }
+								onChange={ this.onChangeDay }
+								onBlur={ this.updateDay }
+							/>
+						</div>
+						<div key="year-field" className="components-datetime__time__form__container components-datetime__time__form__container--year">
+							<input
+								aria-label={ __( 'Year' ) }
+								type="number"
+								name="year"
+								step={ 1 }
+								value={ year }
+								onChange={ this.onChangeYear }
+								onBlur={ this.updateYear }
+							/>
+						</div>
+					</div>
+				</fieldset>
+
+				<fieldset>
+					<legend className="invisible">{ __( 'Time' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--time">
+							<input
+								aria-label={ __( 'Hours' ) }
+								type="number"
+								name="hours"
+								step={ 1 }
+								min={ this.minHours }
+								max={ this.maxHours }
+								value={ hours }
+								onChange={ this.onChangeHours }
+								onBlur={ this.updateHours }
+							/>
+							<span
+								className="components-datetime__time__separator"
```

Comment:
This separator is technically locale-specific, no? Getting this to work I think would depend on #9846, but there are locales that use, for instance, `14h35` instead of `14:35`. I'd think we'd eventually want to translate this, might be worth a comment 🤷‍♂️

Reply:
This is reused from the older version. I decided not to touch this bit that much anyway.

Reply:
This is based on the old codebase and is not locale based. (I wanted to change it to behave that way, but it would overcomplicate things and go way past the scope of the original ticket.)

So what happens is looks up the `a` in the time display string from the WP backend to determine if we are using a 12 or 24 hour clock.

See: https://github.com/aldavigdis/gutenberg/blob/fix_datepicker_a11y/editor/components/post-schedule/index.js#L15